fem/coefficient.hpp

```diff
@@ -946,6 +946,71 @@ class OuterProductCoefficient : public MatrixCoefficient
                      const IntegrationPoint &ip);
 };
 
+class QuadratureFunction;
+
+/// Quadrature function vector coefficient
+class VectorQuadratureFunctionCoefficient : public VectorCoefficient
+{
+private:
+   QuadratureFunction *QuadF; //do not own
+   int index;
+   int length;
+
+public:
+   /// constructor with a quadrature function as input
+   VectorQuadratureFunctionCoefficient(QuadratureFunction *qf);
+
+   /// constructor with a null qf
+   VectorQuadratureFunctionCoefficient() : VectorCoefficient(0), QuadF(NULL),
+      index(-1), length(0) {}
+
+   void SetQuadratureFunction(QuadratureFunction *qf);
+
+   /// set the starting index within the QuadFunc that'll be used to project outwards
+   /// if length is set to a value which will go out of bounds after this is changed than
+   /// it will be changed so that things still work. You should always change length right
+   /// after this is changed.
```

Comment:
I think SetComponent and SetNumComponents are better names. These should probably be combined in one function, as they must always be set together. Capitalized sentences would be good too, in the same doxygen style as in the other hpp.

Reply:
I'm fine with combining these functions. If we were to modify the name for these combined functions I'd be okay with it just being SetComponent.